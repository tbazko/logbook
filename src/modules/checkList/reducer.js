import _ from 'lodash';
import * as types from './constants'

const initialState = {
  items: null,
  logs: null,
}

// const initialState = {
//   items: {
//     [id]: { title, descr }
//   },
//   logs: {
//     [timestamp]: {
//       [id]: { completed }
//     }
//   }
// }

export default (state = initialState, action) => {
  const { type } = action;
  const logs = {};

  switch (type) {
    case types.ADD_ITEM:
      return {
        ...state,
        items: {
          ...state.items,
          [action.id]: { title: action.title },
        },
      }
    case types.TOGGLE_ITEM_CHECKBOX:
      return {
        ...state,
        logs: {
          ...state.logs,
          [action.timestamp]: {
            ..._.get(state, `logs[${action.timestamp}]`, undefined),
            [action.id]: {
              completed: !_.get(state, `logs[${action.timestamp}][${action.id}].completed`, true),
            },
          },
        },
      }
    case types.REMOVE_LIST_ITEM:
      return {
        ...state,
        items: _.omit(state.items, [`${action.id}`]),
      }
    // case types.CHECK_ALL:
    //   return state.map(item => ({ ...item, completed: true }))
    case types.SET_DEFAULT_ITEMS:
    case types.UNCHECK_ALL:
      if (state.logs && state.logs[action.timestamp]) return state

      if (!state.items) return { ...state, logs: null }
      Object.keys(state.items).forEach((item) => {
        logs[item] = { completed: false }
      });

      return {
        ...state,
        logs: {
          ...state.logs,
          [action.timestamp]: logs,
        },
      }
    // case types.REMOVE_ALL:
    //   return []

    default:
      return state
  }
};
