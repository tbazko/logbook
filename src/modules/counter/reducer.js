import * as types from './constants';


export const initialState = {
  value: 0,
};

export default (state = initialState, action) => {
  const { type } = action;

  switch (type) {
    case types.RESET_COUNTER:
      return {
        value: 0,
      };
    case types.DECREMENT_COUNTER:
      return {
        value: state.value - 1,
      };
    case types.INCREMENT_COUNTER:
      return {
        value: state.value + 1,
      };
    default:
      return state;
  }
};
