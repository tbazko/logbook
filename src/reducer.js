import { combineReducers } from 'redux'
import * as reducers from 'reducers'

const rootReducer = combineReducers({
  activityTypes: reducers.activityTypes,
  historicalActivityTypes: reducers.historicalActivityTypes,
  validationError: reducers.validationError,
  activityLogs: reducers.activityLogs,
  activeDateTimestamp: reducers.activeDateTimestamp,
})

export default rootReducer
