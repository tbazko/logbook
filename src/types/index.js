export const ActivityCheckListHeader = {
  SHOW_NEXT_DATE: 'logbook.ActivityCheckListHeader.SHOW_NEXT_DATE',
  SHOW_PREV_DATE: 'logbook.ActivityCheckListHeader.SHOW_PREV_DATE',
  SET_VIEW_MODE: 'logbook.ActivityCheckListHeader.SET_VIEW_MODE',
}

export const AddActivityForm = {
  ADD_ACTIVITY_TYPE: 'logbook.AddActivityForm.ADD_ACTIVITY_TYPE',
  ADD_ACTIVITY_ERROR: 'logbook.ADD_ACTIVITY_ERROR',
}

export const ActivityCheckList = {
  SET_ACTIVE_CHECKLIST: 'logbook.ActivityCheckList.SET_ACTIVE_CHECKLIST',
  TOGGLE_ITEM_CHECKBOX: 'logbook.ActivityCheckList.TOGGLE_ITEM_CHECKBOX',
  REMOVE_LIST_ITEM: 'logbook.ActivityCheckList.REMOVE_LIST_ITEM',
  REMOVE_ALL: 'logbook.ActivityCheckList.REMOVE_ALL',
  SET_DEFAULT_VALUES: 'logbook.ActivityCheckList.SET_DEFAULT_VALUES',
  CHECK_ALL: 'logbook.ActivityCheckList.CHECK_ALL',
  UNCHECK_ALL: 'logbook.ActivityCheckList.UNCHECK_ALL',
}

export const Debug = {
  INSERT_DATA: 'logbook.Debug.INSERT_DATA',
}
