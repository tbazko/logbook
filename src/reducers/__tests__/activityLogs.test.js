import * as types from 'types'
import reducer from '../activityLogs'

describe('CheckList Reducer', () => {
  let state;

  it('should return null', () => {
    state = reducer(state, {
      type: 'TEST',
    });
    expect(state).toMatchSnapshot();
  });

  it('should save objects "2" and "3" in "activityLogs" under "timestamp-1"', () => {
    state = reducer(state, {
      type: types.ActivityCheckList.TOGGLE_ITEM_CHECKBOX,
      id: '2',
      timestamp: 'timestamp-1',
    });
    state = reducer(state, {
      type: types.ActivityCheckList.TOGGLE_ITEM_CHECKBOX,
      id: '3',
      timestamp: 'timestamp-1',
    });
    expect(state).toMatchSnapshot();
  });

  it('should save object "2" in "activityLogs" under timestamp "timestamp-2"', () => {
    state = reducer(state, {
      type: types.ActivityCheckList.TOGGLE_ITEM_CHECKBOX,
      id: '2',
      timestamp: 'timestamp-2',
    });
    expect(state).toMatchSnapshot();
  });

  // it('should set completed for all elements to true', () => {
  //   state = reducer(state, {
  //     type: types.CHECK_ALL,
  //   });
  //   expect(state).toMatchSnapshot();
  // });

  // it('should set completed for all elements to false', () => {
  //   state = reducer(state, {
  //     type: types.UNCHECK_ALL,
  //   });
  //   expect(state).toMatchSnapshot();
  // });

  it('should set completed for all elements in "activityLogs" under "timestamp-3" to "false"', () => {
    const emptyLogsState = reducer({ ...state, activityLogs: null }, {
      type: types.ActivityCheckList.SET_DEFAULT_VALUES,
      timestamp: 'timestamp-3',
    });
    expect(emptyLogsState).toMatchSnapshot();
  });
});
