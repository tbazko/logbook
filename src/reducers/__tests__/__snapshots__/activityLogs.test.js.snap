// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckList Reducer should return null 1`] = `null`;

exports[`CheckList Reducer should save object "2" in "activityLogs" under timestamp "timestamp-2" 1`] = `
Object {
  "timestamp-1": Object {
    "2": Object {
      "completed": false,
    },
    "3": Object {
      "completed": false,
    },
  },
  "timestamp-2": Object {
    "2": Object {
      "completed": false,
    },
  },
}
`;

exports[`CheckList Reducer should save objects "2" and "3" in "activityLogs" under "timestamp-1" 1`] = `
Object {
  "timestamp-1": Object {
    "2": Object {
      "completed": false,
    },
    "3": Object {
      "completed": false,
    },
  },
}
`;

exports[`CheckList Reducer should set completed for all elements in "activityLogs" under "timestamp-3" to "false" 1`] = `
Object {
  "activityLogs": null,
  "timestamp-1": Object {
    "2": Object {
      "completed": false,
    },
    "3": Object {
      "completed": false,
    },
  },
  "timestamp-2": Object {
    "2": Object {
      "completed": false,
    },
  },
}
`;
